stage #3/light-talks/spring-best-practices/spring-react/wisdom-pet/src/main/java/com/frankmoley/lil/wisdompet/services/ServiceService.java

```diff
@@ -0,0 +1,58 @@
+package com.frankmoley.lil.wisdompet.services;
+
+import com.frankmoley.lil.wisdompet.data.entities.ServiceEntity;
+import com.frankmoley.lil.wisdompet.data.repositories.ServiceRepository;
+import com.frankmoley.lil.wisdompet.web.errors.NotFoundException;
+import com.frankmoley.lil.wisdompet.web.models.Service;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@org.springframework.stereotype.Service
+public class ServiceService {
+
+  private final ServiceRepository serviceRepository;
+
+  public ServiceService(ServiceRepository serviceRepository) {
+    this.serviceRepository = serviceRepository;
+  }
+
+  public List<Service> getAllServices(){
+    Iterable<ServiceEntity> entities = this.serviceRepository.findAll();
+    List<Service> services = new ArrayList<>();
+    entities.forEach(entity -> {
+      services.add(this.translateDbToWeb(entity));
+    });
+    return services;
+  }
+
+  public Service getService(long id){
+    Optional<ServiceEntity> optional = this.serviceRepository.findById(id);
```

Comment:
I would probably do
`this.serviceRepository.findById(id).map(this::translateDbToWeb).orElseThrow(() -> ...); `